projects/igniteui-angular/src/lib/chips/chips-area.spec.ts

```diff
@@ -478,6 +481,15 @@ describe('IgxChipsArea', () => {
         secondChipComp.chipArea.nativeElement.dispatchEvent(keyEvent);
         fix.detectChanges();
         expect(secondChipComp.onSelection.emit).toHaveBeenCalled();
+
+        let chipsSelectionStates = fix.componentInstance.chips.toArray().filter(c => c.selected);
+        expect(chipsSelectionStates.length).toEqual(1);
```

Comment:
It would be better to check what the `onSelection` event returns using `toHaveBeenCalledWith`. For example:
```
expect(secondChipComp.onSelection.emit).toHaveBeenCalledWith({
  owner: chipAreaComp,
  newSelection: [secondChipComp]
});
```

Same with the other places I suggested to check what `onSelection` returns.